rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -39,21 +39,23 @@ import monix.eval.Coeval
   */
 trait Reduce[M[_]] {
 
-  def produce(chan: Quote, data: Seq[Par], persistent: Boolean)(implicit env: Env[Par],
-                                                                rand: Blake2b512Random): M[Unit]
-
-  def consume(binds: Seq[(BindPattern, Quote)], body: Par, persistent: Boolean)(
-      implicit env: Env[Par],
-      rand: Blake2b512Random): M[Unit]
+  protected[this] implicit def monadM: Monad[M]
```

Comment:
We don't need to put this constraint in the trait. It should work with any `M[_]` at this point.